### PR TITLE
Add net7.0 to JsonCli and remove net3.1

### PR DESCRIPTION
**Pull request by nturinski**
Adding the configurations to add .net7 templates

### CODE REVIEW

**Review by bwateratmsft — DISMISSED** [review]

File: src/templates/dotnet/executeDotnetTemplateCommand.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ async function getFramework(context: IActionContext, workingDirectory: string |
         }
 
         // Prioritize "LTS", then "Current", then "Preview"
-        const netVersions: string[] = ['6.0', '5.0', '3.1'];
+        const netVersions: string[] = ['7.0', '6.0', '5.0', '3.1'];
```
Comment (bwateratmsft) [inline src/templates/dotnet/executeDotnetTemplateCommand.ts:64]: Same feedback as on #3472, based on the comment this should be 6.0, 7.0, 5.0 (6.0 is an LTS release). And 3.1 should be removed.

**Comment by nturinski** [comment]
Waiting to get this build working https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build/results?buildId=7100406&view=results

Currently still investigating

**Comment by nturinski** [comment]
Using this PRs as references:
https://github.com/microsoft/vscode-azurefunctions/pull/3472
https://github.com/microsoft/vscode-azurefunctions/pull/2922

File: .azure-pipelines/common/test.yml [hunk 2]
```diff
@@ -1,47 +1,48 @@
 steps:
-- script: |
```
Comment (alexweininger) [inline .azure-pipelines/common/test.yml:2]: Can you revert the whitespace changes in this file?

File: package.json [hunk 3]
```diff
@@ -1060,7 +1060,7 @@
         "@microsoft/eslint-config-azuretools": "^0.1.0",
         "@microsoft/vscode-azext-dev": "^0.1.4",
         "@types/fs-extra": "^8.1.0",
-        "@types/gulp": "^4.0.6",
```
Comment (alexweininger) [inline package.json:1063]: Just curious, why was this needed?

**Review by alexweininger — APPROVED** [review]

**Review by alexweininger — DISMISSED** [review]

**Comment by nturinski** [comment]
image

File: package.json (hunk 3 above)
Comment (nturinski) [reply package.json:1063]: I think I ran an npm audit it just auto updated it. There were security vulnerabilities that I didn't quite understand.

File: .azure-pipelines/common/test.yml (hunk 2 above)
Comment (nturinski) [reply .azure-pipelines/common/test.yml:2]: Ugh my stupid auto-formatter...

**Review by alexweininger — APPROVED** [review]
